src/ptvsd/adapter/messages.py

```diff
@@ -147,7 +150,8 @@ def event(self, event):
 
     # Generic request handler, used if there's no specific handler below.
     def request(self, request):
-        return self._server.delegate(request)
+        if self._no_debug is False:
```

Comment:
This needs to fail the requests otherwise. As it is, it will crash the message loop, because the handler didn't respond to a request.

Actually, come to think of it, this is also the case if the IDE sends us some request we don't know about, before we actually connect to the server - even if it's not a nodebug scenario. So this should check `_channels.server`, and delegate if it's there, and fail the request otherwise.